nio/client/base_client.py

```diff
@@ -120,9 +121,15 @@ def inner(self, *args, **kwargs):
 class ClientCallback:
     """nio internal callback class."""
 
-    func: Callable = field()
+    func: Union[Callable[..., None], Callable[..., Awaitable[None]]] = field()
```

Comment:
TODO: More complete typing here, potentially obsolete the `ResponseCallback` class.